### ISSUE TITLE
Inline field editor doesn't preserve line breaks

### ISSUE THREAD

**Issue by breadcat** [salesagility/SuiteCRM#3769]
#### Issue
When editing a custom TextArea field (may also affect built in fields, haven't tested), using the inline editor will allow you to edit the field and save the contents however if you previously had line breaks in the field they won't br preserved

#### Expected Behavior
Line breaks in the field should be preserved.

#### Actual Behavior
Line breaks aren't preserved and the resulting text is just one long string.

#### Possible Fix
Use the older non-inline editor section, this works without issue.

#### Steps to Reproduce
1. Double click a textarea field with a line break in it
2. Edit the text
3. Press enter to save the boxes contents
4. Line breaks are replaced

#### Context
* Low priority

#### Your Environment
* SuiteCRM Version used:  7.9.1
* Browser name and version: Firefox 54.0 (32-bit)
* Environment name and version: Debian 9, PHP5.4.36, MySQL5.5.40
* Operating System and version: Windows 7 x86

**Comment by Mausino** [comment]
@breadcat is your issue duplicate of this issue #3136 ???  there is also a patch

from which version did you upgrade the SuiteCRM  ??? or it is fresh install ???

**Comment by breadcat** [comment]
Thanks @Mausino, yes it does appear to be a duplicate. I only searched in open issues, apologies.

I upgraded my copy of SuiteCRM from 7.1.x (not sure of exact version) using:
SuiteCRM-Upgrade-7.1.x-to-7.3.2 > 7.3.x-to-7.4.4 > 7.4.x-to-7.6.10 > 7.6.x-to-7.8.5 > 7.8.x-to-7.9.1

Thanks for your help.

**Comment by pgorod** [comment]
@breadcat how come you don't have the #3136 fix in your system? 

Or do you have the fix, but it's not solving your specific case of the bug?

**Comment by breadcat** [comment]
I do have this file, and the commit 4f431263e11993323546c488c09b02a7495fa5ca is in present as well.
This is a little bit strange, I can't shift-enter either to add a newline, it submits the textarea as it would when you press enter.

I'm wondering if something has altered the permissions on this file, as I needed root access to enter `include/InlineEditing/`, but not `include/`.
Are there any functions in SuiteCRM itself to repair (or check) permissions?

**Comment by pgorod** [comment]
You can go into Admin/upgradeWizard and run the first step, it checks permissions. You don't have to complete any upgrade.

This is not a full permissions check,m but might give you some clues. Otherwise just go into the server with SSH or FTP and check permissions _and ownership_ of those files.

**Comment by breadcat** [comment]
Okay, I've checked the permissions for the directory using the upgrade wizard and corrected some ownership issues (root as owner instead of suitecrm user) issues, however the original issue is still there.

Are there any logs, or files I can provide you with to help narrow anything down?

**Comment by breadcat** [comment]
I've checked through crontab for both users, and nothing looks untowards there.
I always SSH in as my non-root user, and will generally only `sudo` when performing package updates or controlling services, etc. The correct permissions and owners are now in place too.

I've gone through the process of editing a textarea and have copied the logs below. Nothing of real merit as far as I'm concerned: https://pastebin.com/znA8D3dM

The edit happens around line 360.

**Comment by pgorod** [comment]
I don't know...
Do you see any difference if you try it in a non-custom field, or if you try it in the online demo?

**Comment by breadcat** [comment]
Okay, this is getting stranger.

I tried using your [demo](https://demo.suiteondemand.com), and it worked fine from my home computer (Firefox 54 64-bit, Windows 10 64-bit) then tried accessing my own SuiteCRM install from home which also worked flawlessly.

Must be an issue with the computers at work, however I tested it using Chrome there as well just in case it was a specific browser. I'll try clearing caches etc tomorrow and see if that makes any difference.
